src/CommandProcessor.php

```diff
@@ -137,8 +141,19 @@ public function validateRunAndAlter(
             return $validated;
         }
 
-        // Run the command, alter the results, and then handle output and status
-        $result = $this->runCommandCallback($commandCallback, $commandData);
+        $replaceDispatcher = new ReplaceCommandHookDispatcher($this->hookManager(), $names);
+        if ($this->logger) {
+            $replaceDispatcher->setLogger($this->logger);
+        }
+        if ($replaceDispatcher->hasReplaceCommandHook()) {
```

Comment:
Why did you add this extra code? It was much nicer the way it was before. A command replacement hook can only every replace one command, so the function arguments will always be well-known. Just use the same signature as the calling function, per your original implementation. The code is cleaner, and the UX is better.